gomock/doc_test.go

```diff
@@ -0,0 +1,66 @@
+package gomock_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	mock_sample "github.com/golang/mock/sample/mock_user"
+)
+
+func ExampleCall_DoAndReturn() {
+	t := &testing.T{} // provided by test
+	ctrl := gomock.NewController(t)
+	mockIndex := mock_sample.NewMockIndex(ctrl)
+
+	mockIndex.EXPECT().Get(gomock.Any()).DoAndReturn(
+		func() string {
+			time.Sleep(1 * time.Second)
+			return "I'm sleepy"
+		},
+	)
+}
+
+func ExampleCall_DoAndReturn_captureArguments() {
+	t := &testing.T{} // provided by test
+	ctrl := gomock.NewController(t)
+	mockIndex := mock_sample.NewMockIndex(ctrl)
+	var s string
+
+	mockIndex.EXPECT().Get(gomock.AssignableToTypeOf(s)).DoAndReturn(
+		// When capturing arguments the anonymous function should have the same signature as the mocked method.
+		func(arg string) interface{} {
+			time.Sleep(1 * time.Second)
```

Comment:
nit: remove for capture example. Also, maybe make Millisecond if these will be run with tests to keep things snappy.